task4/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type OutputResult struct {
+	totalTimeReq    int64
+	averageTimeReq  int64
+	maxResponseTime int64
+	minResponseTime int64
+	countMissResp   int
+}
+
+type InputParameters struct {
+	url        string
+	countOfReq int
+	timeOut    int
+}
+
+func sendReq() (out OutputResult, err error) {
+	parameters := initParameters()
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+	var totalReqTime int64
+	var countReq int64
+	var allTimeResp []int64
+	var countMissResp int
+
+	for i := 0; i < parameters.countOfReq; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			start := time.Now()
+			client := http.Client{
+				Timeout: time.Duration(parameters.timeOut),
+			}
+			_, err := client.Get(parameters.url)
+			checkErr(err)
+
+			timeReq := time.Since(start).Nanoseconds()
+			mutex.Lock()
```

Comment:
if it possible - don't use wide lock, only in special place, where it really need